### PR TITLE
Rotate firebase token in case of error

### LINKED ISSUE #3752: Push-Notification Option for recreate token if test failed
Here a dedicated issue of this slice different Problem:
https://github.com/element-hq/element-x-android/issues/3570#issuecomment-2423760827

### Steps to reproduce

1. Login to EXA
2. See in microg if EXA was correct registerd to FCM
3. test push works in EXA
4. Delete registration in microg
5. Run test in EXA again

#### What did you expect?

Token validation success, but loop test failed.

#### What happened instead?
Here are two Option to fix it:
 - Option to reset the token to trigger new FCM registration
 - Reset FCM-Token after change Norification Provider (e.g. to ntfy)

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Add a quick fix for the troubleshoot notification test in case of PusherRejected error, which means that the token is unknown.

In this case, the quick fix will rotate the Firebase token.

Note: maybe later we will see if we can do the same thing for UnifiedPush provider.

## Motivation and context

Closes #3752 

## Screenshots / GIFs

You can see in the video below that when clicking on QuickFix, the token is rotated.

https://github.com/user-attachments/assets/0152344f-809e-482b-9cfa-1e2a6673d54d

## Tests

- @wrenix can you check if this is fixing your issue please?

Internally, this can be tested by patching the code and let `DefaultPushService.testPush()` throw a `PushGatewayFailure.PusherRejected()` (this is what I have done for the video above).

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenDeleter.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * This class deletes the current Firebase token.
+ */
+@ContributesBinding(AppScope::class)
+class DefaultFirebaseTokenDeleter @Inject constructor(
+    private val isPlayServiceAvailable: IsPlayServiceAvailable,
+) : FirebaseTokenDeleter {
+    override suspend fun delete() {
+        suspendCoroutine { continuation ->
+            // 'app should always check the device for a compatible Google Play services APK before accessing Google Play services features'
+            if (isPlayServiceAvailable.isAvailable()) {
```
Comment (jmartinesp) [inline libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenDeleter.kt:33]: Not a needed change, but an early return here with `!isPlayServiceAvailable.isAvailable()` might be more readable. Also, for the exception, maybe a `throw Exception(...).also(Timber::e)` might work?

File: libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenRotator.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import io.element.android.libraries.di.AppScope
+import javax.inject.Inject
+
+interface FirebaseTokenRotator {
+    suspend fun rotate(): Result<Unit>
+}
+
+/**
+ * This class delete the Firebase token and generate a new one.
+ */
```
Comment (jmartinesp) [inline libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenRotator.kt:20]: I think the comment should be in the interface instead?

**Review by jmartinesp — APPROVED** [review]
I haven't tried it yet, but the code LGTM, just a couple of suggestions.

File: libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenDeleter.kt (hunk 1 above)
Comment (bmarty) [reply libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenDeleter.kt:33]: I will change. I am also wondering why we throw the exception here and we do not invoke `continuation.resumeWithException(e)` 🤔

Comment (bmarty) [reply libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenDeleter.kt]: 12f5839eefa066dce490a40775bbdaf01f902076

File: libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenRotator.kt (hunk 2 above)
Comment (bmarty) [reply libraries/pushproviders/firebase/src/main/kotlin/io/element/android/libraries/pushproviders/firebase/FirebaseTokenRotator.kt:20]: Yes, thanks! 12f5839eefa066dce490a40775bbdaf01f902076

**Pull request merged**